File: content/snippet/code-signal/arcade/intro/smooth-sailing/sort-by-height/sort-by-height.py
def sort_by_height(a):
    for i in range(len(a)):
        for j in range(i+1, len(a)):
            if a[i] != -1 and a[j] != -1 and a[i] > a[j]:
                a[i], a[j] = a[j], a[i]
    return a


print(sort_by_height([-1, 150, 190, 170, -1, -1, 160, 180]))

File: content/snippet/code-signal/arcade/intro/smooth-sailing/sort-by-height/sort-by-height.js
function sortByHeight(a) {
  const sorted = a.filter((x) => x !== -1).sort((x, y) => x - y);
  return a.map((x) => (x === -1 ? x : sorted.shift()));
}

console.log(sortByHeight([-1, 150, 190, 170, -1, -1, 160, 180]));
